02_oop/src/mediaplayer/player/playergui.cpp

```diff
@@ -0,0 +1,288 @@
+	}
+	
+	void PlayerGui::getFile(std::shared_ptr<MediaFile> file) {
+		mCurrPlayFile = file;
+		if (!file)
+			return;
+		auto status = mpMediaPlayer->state();
+		mpMediaPlayer->setMedia(QUrl::fromLocalFile(QString::fromStdString(mCurrPlayFile->path())));
+		if (status != QMediaPlayer::StoppedState)
+			play();
+	}
+	
+	void PlayerGui::getFileForPlay(std::shared_ptr<MediaFile> file) {
+		mCurrPlayFile = file;
+		if (!file)
+			return;
+		play();
+	}
+	
+	void PlayerGui::openPlaylist() {
+		mpPlaylist->isVisible() ? mpPlaylist->hide() : mpPlaylist->show();
+	}
+	
+	void PlayerGui::openLibrary() {
+		mpLibrary->isVisible() ? mpLibrary->hide() : mpLibrary->show();
+	}
+	
+	bool PlayerGui::play() {
+		if (mCurrPlayFile) {
+			mpMediaPlayer->setMedia(QUrl::fromLocalFile(QString::fromStdString(mCurrPlayFile->path())));
+			mpMediaPlayer->play();
+			if (mpMediaPlayer->state() == QMediaPlayer::PlayingState) {
+				mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPause));
+				mpShowMedia->setText(QString::fromStdString(mCurrPlayFile->path()));
+				return true;
+			}
+		}
+		mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPlay));
+		return false;
+	}
+	
+	void PlayerGui::slotPlay() {
+		switch(mpMediaPlayer->state()) {
+			case QMediaPlayer::PlayingState:
+				mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPlay));
+				mpMediaPlayer->pause();
+				break;
+			case QMediaPlayer::PausedState:
+				mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPause));
+				mpMediaPlayer->play();
+				break;
+			case QMediaPlayer::StoppedState:
+				if (!play())
+					emit playCurrentFile();
+				break;
+		}
+	}
+
+	void PlayerGui::slotStop() {
+		mpMediaPlayer->stop();
+		mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPlay));
+	}
+	
+	void PlayerGui::slotSetMediaPosition(int n) {
+		mpMediaPlayer->setPosition(n);
+	}
+	
+	QString PlayerGui::msecsToString(qint64 sec) {
+		int hours = (sec / (60 * 60 * 1000));
+		int minutes = ((sec % (60 * 60 * 1000)) / (60 * 1000));
+		int seconds = ((sec % (60 * 1000)) / 1000);
+		return QTime(hours, minutes, seconds).toString("hh:mm:ss");
+	}
+	
+	void PlayerGui::slotSetSliderPosition(qint64 pos) {
+		mProgress->setValue(pos);
+		mTime->setText(msecsToString(pos));
+		int nDuration = mProgress->maximum();
+		mRestTime->setText(msecsToString(nDuration - pos));
+	}
+	
+	void PlayerGui::slotSetDuration(qint64 n) {
+		mProgress->setRange(0, n);
+		mTime->setText(msecsToString(0));
+		mRestTime->setText(msecsToString(n));
+	}
+	
+	void PlayerGui::slotSetMute() {
+		static bool isMute = false;
+		if (isMute) {
+			emit mpMediaPlayer->setMuted(!isMute);
+			mVolumeMute->setIcon(style()->standardIcon(QStyle::SP_MediaVolume));
+			isMute = false;
+		} else {
+			emit mpMediaPlayer->setMuted(!isMute);
+			mVolumeMute->setIcon(style()->standardIcon(QStyle::SP_MediaVolumeMuted));
+			isMute = true;
+		}
```

Comment:
Looks like can be simplified. Actions are pretty similar so condition not needed.
```c++
emit mpMediaPlayer->setMuted(!isMute);
QStyle icon = isMute ? QStyle::SP_MediaVolume : QStyle::SP_MediaVolumeMuted;
mVolumeMute->setIcon(style()->standardIcon(icon));
isMute = !isMute;

Reply:
Maybe even update `isMute` firstly and use after.
``` c++
isMute = !isMute;
emit mpMediaPlayer->setMuted(isMute);
QStyle icon = isMute ? QStyle::SP_MediaVolumeMuted :: QStyle::SP_MediaVolume;
mVolumeMute->setIcon(style()->standardIcon(icon));
```